### PR TITLE
Support big-endian on the zlib's header

### PR DESCRIPTION
**Pull request by dinosaure**
/cc @talex5 this patch should fix one error about big-endian - but may be we have some others bugs.

### CODE REVIEW

**Comment by dinosaure** [comment]
This PR fix the problem about big-endian support. It should help/fix `ocaml-git` big-endian support too.

**Comment by dinosaure** [comment]
thx @talex5 to give me an access to a `s390x` machine.

File: lib/de.ml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 [@@@landmark "auto"]
 
+let () = Printexc.record_backtrace true
```
Comment (talex5) [inline lib/de.ml:3]: Left-over debugging?

Comment (dinosaure) [reply lib/de.ml:3]: ah yes

**Pull request merged**